Automattic/jetpack
# Boost: fix comments and error reporting in Cache settings

In #35249 I forgot to fix 2 docblock comments, and I didn't handle error reporting correctly when creating the config file for the first time, so those are fixed here.

Similarly, I didn't handle return values from Boost_Cache_Utils::write_to_file() so that is fixed here too.

Logging of errors is done to the internal variable, "last_error", but there was no way for outside code to reset that variable after reading it. This might result in the same error being reported multiple times. Whenever the last_error is read, it should be reset.
Should that be done automatically?

### Other information:


## Jetpack product discussion
pc9hqz-2vF-p2

## Does this pull request change what data or activity we track or use?
no

## Testing instructions:
* Delete wp-content/boost-cache/ and make it again with your own user so the web server can't write to it.
* Add an error_log on line 48 of Boost_Cache_Settings.php to alert you if the write fails.
* Open your test site and check the error_log. Hopefully you'll see your error_log message there when it can't create the config file.
